src/nvim/window.c

```diff
@@ -3972,28 +3972,31 @@ void free_tabpage(tabpage_T *tp)
 /// @param after Put new tabpage after tabpage "after", or after the current
 ///              tabpage in case of 0.
 /// @param filename Will be passed to apply_autocmds().
-/// @return Was the new tabpage created successfully? FAIL or OK.
-int win_new_tabpage(int after, char *filename)
+/// @return Pointer to the new tabpage, or NULL if error.
+tabpage_T *win_new_tabpage(int after, char *filename, bool enter)
```

Comment:
The need for `enter` here is a sign that this API is more than just a wrapper, it's actually achieving new functionality. Right?

Reply:
Yes, since this would allow creating a new tabpage without entering it. Not sure if there's already a way to do that via Lua though, or if there's really even a use case for it.